cmake/test/gtest.cmake

```diff
@@ -85,97 +85,261 @@ function(catkin_add_executable_with_gtest target)
-    endif()
-    find_file(_CATKIN_GTEST_INCLUDE "gtest.h"
-      PATHS ${_paths}
-      NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH)
-
-    if(_CATKIN_GTEST_SRC)
-      get_filename_component(_CATKIN_GTEST_SOURCE_DIR ${_CATKIN_GTEST_SRC} PATH)
-      get_filename_component(_CATKIN_GTEST_BASE_DIR ${_CATKIN_GTEST_SOURCE_DIR} PATH)
-      # add CMakeLists.txt from gtest dir
-      set(_CATKIN_GTEST_BINARY_DIR ${CMAKE_BINARY_DIR}/gtest)
-
-      # overwrite CMake install command to skip install rules for gtest targets
-      # which have been added in version 1.8.0
-      set(_CATKIN_SKIP_INSTALL_RULES TRUE)
-      function(install)
-        if(_CATKIN_SKIP_INSTALL_RULES)
-          return()
-        endif()
-        _install(${ARGN})
-      endfunction()
-      add_subdirectory(${_CATKIN_GTEST_BASE_DIR} ${_CATKIN_GTEST_BINARY_DIR})
-      set(_CATKIN_SKIP_INSTALL_RULES FALSE)
-
-      # mark gtest targets with EXCLUDE_FROM_ALL to only build when tests are built which depend on them
-      set_target_properties(gtest gtest_main PROPERTIES EXCLUDE_FROM_ALL 1)
-      get_filename_component(_CATKIN_GTEST_INCLUDE_DIR ${_CATKIN_GTEST_INCLUDE} PATH)
-      get_filename_component(_CATKIN_GTEST_INCLUDE_DIR ${_CATKIN_GTEST_INCLUDE_DIR} PATH)
-      # set from-source variables
-      set(GTEST_FROM_SOURCE_FOUND TRUE CACHE INTERNAL "")
-      set(GTEST_FROM_SOURCE_INCLUDE_DIRS ${_CATKIN_GTEST_INCLUDE_DIR} CACHE INTERNAL "")
-      set(GTEST_FROM_SOURCE_LIBRARY_DIRS ${_CATKIN_GTEST_BINARY_DIR} CACHE INTERNAL "")
-      set(GTEST_FROM_SOURCE_LIBRARIES "gtest" CACHE INTERNAL "")
-      set(GTEST_FROM_SOURCE_MAIN_LIBRARIES "gtest_main" CACHE INTERNAL "")
-      message(STATUS "Found gtest sources under '${_CATKIN_GTEST_BASE_DIR}': gtests will be built")
-    endif()
-    if(NOT GTEST_FROM_SOURCE_FOUND)
+#
+# Find GTest source-only install
+#
+# Google recommends distributing GTest as source only, to be built with the same
+# flags as that which is being tested.
+#
+# :param[in] include_paths: Paths to search for GTest includes
+# :param[in] src_paths: Paths to search for GTest sources
+# :param[out] found: Whether or not GTest was found in the paths provided
+# :param[out] base_dir: The base directory containing GTest's CMakeLists.txt
+# :param[out] include_dir: The include path to access GTest's headers
+# :param[out] lib_dir: The library path to access GTest's libraries
+# :param[out] libs: GTest's libraries
+# :param[out] main_libs: GTest's main libraries
+#
+# @public
+#
+function(catkin_find_gtest_source include_paths src_paths found base_dir
+         include_dir lib_dir libs main_libs)
+  # Find the gtest headers
+  find_file(_GTEST_INCLUDES "gtest.h"
+            PATHS ${include_paths}
+            NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH
+  )
+
+  # Find the gtest sources
+  find_file(_GTEST_SOURCES "gtest.cc"
+           PATHS ${src_paths}
+           NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH
```

Comment:
The indentation here seems off by a space.

Reply:
Nice catch, fixed both.